lib/features/occupation/presentation/pages/change_occupation_page.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:coffeecard/base/strings.dart';
+import 'package:coffeecard/features/occupation/presentation/cubit/occupation_cubit.dart';
+import 'package:coffeecard/features/occupation/presentation/widgets/occupation_form.dart';
+import 'package:coffeecard/features/user/presentation/cubit/user_cubit.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:coffeecard/widgets/components/loading.dart';
+import 'package:coffeecard/widgets/components/scaffold.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ChangeOccupationPage extends StatelessWidget {
+  const ChangeOccupationPage();
+
+  static Route get route =>
+      MaterialPageRoute(builder: (_) => const ChangeOccupationPage());
+
+  @override
+  Widget build(BuildContext context) {
+    return AppScaffold.withTitle(
+      title: Strings.changeOccupation,
+      body: BlocProvider(
+        create: (_) => sl<OccupationCubit>()..fetchOccupations(),
```

Comment:
i think its bad practice to instantiate cubits in UI if it can be avoided. If you want to instantiate the cubit here you need a reference to a usecase which needs a reference to a data source which needs a reference to the API making a UI component dependent on the backend, smells fishy to me. 

This way we leave instantiation to the service locator which already has all the dependencies and simply scope it. But then you may ask, why not just instantiate the cubit here and scope the dependencies from the service locator? well then you can just register the cubit as a factory and scope it this way, now you dont need to even reference the cubits dependencies

Reply:
We had some problems in the past using the service locator for our blocs, though looking into it, it seems that stemmed from passing the bloc using the value parameter instead of the create since this changes the life-time of the bloc. 

As long as we keep registering it as factories in the service locator and passing using the create argument, I guess it is fine to do it in this manner.